ui/src/spa_index.jsx

```diff
@@ -0,0 +1,509 @@
+import CircularProgress from '@material-ui/core/CircularProgress'
+import IconButton from '@material-ui/core/IconButton'
+import Link from '@material-ui/core/Link'
+import Paper from '@material-ui/core/Paper'
+import Typography from '@material-ui/core/Typography'
+import RefreshIcon from '@material-ui/icons/Refresh'
+import SettingsIcon from '@material-ui/icons/Settings'
+import { makeStyles } from '@material-ui/styles'
+import { filter, find, forEach, get, isArray, sortBy, toArray, trim } from 'lodash'
+import React, { useEffect, useState } from 'react'
+import ReactDOM from 'react-dom'
+import { useLocation, useHistory } from 'react-router';
+
+import AllocationReport from './components/AllocationReport';
+import Controls from './components/Controls';
+import DetailsDialog from './components/DetailsDialog';
+import Header from './components/Header';
+import Page from './components/Page';
+import Subtitle from './components/Subtitle';
+import Warnings from './components/Warnings';
+import AllocationService from './services/allocation';
+import { cumulativeToTotals, rangeToCumulative } from './util';
+
+const windowOptions = [
+  { name: 'Today', value: 'today' },
+  { name: 'Yesterday', value: 'yesterday' },
+  { name: 'Week-to-date', value: 'week' },
+  { name: 'Month-to-date', value: 'month' },
+  { name: 'Last week', value: 'lastweek' },
+  { name: 'Last month', value: 'lastmonth' },
+  { name: 'Last 7 days', value: '6d' },
+  { name: 'Last 30 days', value: '29d' },
+  { name: 'Last 60 days', value: '59d' },
+  { name: 'Last 90 days', value: '89d' },
+]
+
+const aggregationOptions = [
+  { name: 'Cluster', value: 'cluster' },
+  { name: 'Node', value: 'node' },
+  { name: 'Namespace', value: 'namespace' },
+  { name: 'Controller kind', value: 'controllerKind' },
+  { name: 'Controller', value: 'controller' },
+  { name: 'Service', value: 'service' },
+  { name: 'Pod', value: 'pod' },
+]
+
+const idleOptions = [
+  { name: 'Hide', value: "hide" },
+  { name: 'Share', value: "share" },
+  { name: 'Separate', value: "separate" },
+]
+
+const accumulateOptions = [
+  { name: 'Entire window', value: true },
+  { name: 'Daily', value: false },
+]
+
+const shareSplitOptions = [
+  { name: 'Share evenly', value: 'even' },
+  { name: 'Share weighted by cost', value: 'weighted' },
+]
+
+const useStyles = makeStyles({
+  reportHeader: {
+    display: 'flex',
+    flexFlow: 'row',
+    padding: 24,
+  },
+  titles: {
+    flexGrow: 1,
+  },
+})
+
+const ReportsPage = () => {
```

Comment:
Trying to remember, because I think we talked about it.. why are we using `.jsx` now? Is this for syntax highlighting, or some functionality? If so, should we convert all files to `.jsx` in the interest of uniformity?

Reply:
I thought it was correcting some minor syntax highlighting at the time, but looking at it again now I don't think it does.

Googled around some and it seems like there's no really great reason to prefer one extension or the other. The most in-depth conversation of it I could find was here: https://github.com/airbnb/javascript/pull/985

It's basically just a bunch of people saying "we should standardize on .js" and one mad Airbnb person being like "Returning inline html elements is not \~javascript\~"

I'm game to just keep `.js` so we don't have to rename a ton of files and because it seems like there's really not a significant difference.

Reply:
Okay, interesting, that's a good find. Just know, I'm open to opinions either way, but I do like keeping things standard, so `.js` is fine with me, but just holler if you want to go `.jsx`.